src/corner_kick/src/types/spritesheet.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * This file specifies types for spritesheet generation
+ */
+
+/**
+ * The various types of shapes that can be drawn in a spritesheet
+ */
+export enum ShapeType {
+    RECT = 'rect',
+    ELLIPSE = 'ellipse',
+    LINE = 'line',
+    ARC = 'arc',
+    POLY = 'poly',
+}
+
+/**
+ * Description object for spritesheet. Used by
+ * `SpritesheetManager` to generate a spritesheet.
+ */
+export interface ISpritesheet {
+    /**
+     * The frames contained in this spritesheet
+     */
+    frames: IFrame[];
+    /**
+     * The size of the spritesheet, make sure all the frames are contained
+     * inside these dimensions
+     */
+    dimensions: {
+        h: number;
+        w: number;
+    };
+}
+
+/**
+ * Describes a single frame in a spritesheet, which may contain many shapes
+ */
+export interface IFrame {
+    /**
+     * The dimension of the frame, any shape drawn outside of these dimensions
+     * will be cropped
+     */
+    dimensions: { x: number; y: number; w: number; h: number };
+    /**
+     * The shapes of this frame, drawn relative to the frame top-left corner
+     */
+    shapes: Array<IRectShape | IEllipseShape | IArcShape | ILineShape | IPolyShape>;
+}
+
+/**
+ * Describes a shape. Used to generate a spritesheet.
+ */
+export interface IShape {
+    /**
+     * Should the shape be filled?
+     */
+    fill?: boolean;
+
+    /**
+     * Should the shape be stroked?
+     */
+    stroke?: boolean;
+
+    /**
+     * The stroke weight of the shape
+     */
+    stroke_weight?: number;
+}
+
+/**
+ * A rectangle shape object. All dimensions are relative to the parent frame's top-left
+ * corner
+ */
+export interface IRectShape extends IShape {
+    type: ShapeType.RECT;
+
+    /**
+     * The data object is specified as follow:
+     * - `data[0]`: x in px
+     * - `data[1]`: y in px
+     * - `data[2]`: width in px
+     * - `data[3]`: height in px
+     */
+    data: [number, number, number, number];
```

Comment:
Why do all the shapes have this `data` member? It doesn't seem like something that's enforced by the `IShape` interface, so I'd expect to see each unique shape have members like `x_in_px`, `y_in_px`, `width_in_px`, `height_in_px`

Reply:
That is because it respects the shape format we agreed upon when designing the earlier communication protocol with the AI. Instead of creating a new protocol, I decided to reuse what we had. + would make future implementation of AI providing the spritesheet format via a service easier.

Reply:
I'm still confused here. Aren't these classes supposed to be abstractions of the various shapes? If we're just representing each shape by an array of numbers we may as well just have one class `Shape` with a `data` member and be done with it?

Reply:
See above. I can definitely do that, you just loose a little bit in typechecking and format definition for each shape format

Reply:
Fixed. We now have a single `IShape` and we specificy what goes in the data field in a JSDoc comment.